dropwizard-lifecycle/src/test/java/io/dropwizard/lifecycle/ServerLifecycleListenerTest.java

```diff
@@ -0,0 +1,113 @@
+package io.dropwizard.lifecycle;
+
+import org.assertj.core.api.Assertions;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ServerLifecycleListenerTest {
+    private static final String APPLICATION = "application";
+    private static final String ADMIN = "admin";
+    private static final String HOST_1 = "192.168.2.5";
+    private static final String HOST_2 = "192.68.4.77";
+    private static final String HOST_3 = "192.168.67.20";
+
+    @Test
+    void getLocalPort() {
+        int localPort = 5673;
+        int adminPort = 12345;
+        Server server = configureAndGetSingleConnectorServer(localPort, adminPort);
+        ServerLifecycleListener listener = (server1) -> {
+        };
+        int retrievedLocalPort = listener.getLocalPort(server);
+        Assertions.assertThat(retrievedLocalPort).isEqualTo(localPort);
+    }
+
+    @Test
+    void getAdminPort() {
+        int localPort = 5673;
+        int adminPort = 12345;
+        Server server = configureAndGetSingleConnectorServer(localPort, adminPort);
+        ServerLifecycleListener listener = (server1) -> {
+        };
+        int retrievedAdminPort = listener.getAdminPort(server);
+        Assertions.assertThat(retrievedAdminPort).isEqualTo(adminPort);
+    }
+
+    private Server configureAndGetSingleConnectorServer(int applicationPort, int adminPort) {
+        Server server = mock(Server.class);
+        ServerConnector applicationConnector = mock(ServerConnector.class);
+        ServerConnector adminConnector = mock(ServerConnector.class);
+        Connector[] connectors = {applicationConnector, adminConnector};
+        when(server.getConnectors()).thenReturn(connectors);
+        configuredServerConnector(applicationConnector, applicationPort, Arrays.asList("ssl", "http/1.1", "http/2"), APPLICATION, HOST_1);
+        configuredServerConnector(adminConnector, adminPort, Arrays.asList("tls", "http/2"), ADMIN, HOST_2);
+        return server;
+    }
+
+    @Test
+    void getPortDescriptorList() {
+        Server server = configureMultiProtocolServer();
+        ServerLifecycleListener listener = (server1) -> {
+        };
+        List<PortDescriptor> portDescriptorList = listener.getPortDescriptorList(server);
+        PortDescriptor[] portDescriptors = buildCompletePortDescriptorsArray();
+        Assertions.assertThat(portDescriptorList).usingElementComparator((o1, o2) -> {
+            if (Objects.equals(o1.getConnectorType(), o2.getConnectorType()) &&
```

Comment:
Perhaps may have been cleaner to just define `PortDescriptor::equals` (and hash). Still looks good 👌